src/connector.c

```diff
@@ -150,36 +215,44 @@ static void on_resolve(uv_getaddrinfo_t *r, int status, struct addrinfo *addr) {
     }
 
     if (status != 0) {
-        cr->cb(cr->sock, status, cr->ctx);
-        tlsuv__free(r);
-    } else {
-        cr->sock = tlsuv_socket(addr, 0);
-        int rc = connect(cr->sock, addr->ai_addr, addr->ai_addrlen);
-        int e = get_error();
+        uv_freeaddrinfo(addrlist);
+        if (cr->cb) cr->cb(cr->sock, status, cr->ctx);
+        free_conn_req(cr);
+        return;
+    }
 
-        if (rc == 0 ||
-#if _WIN32
-            e == WSAEWOULDBLOCK
-#else
-            e == EINPROGRESS
-#endif
-                ) {
-            uv_poll_init_socket(r->loop, &cr->poll, cr->sock);
-            uv_poll_start(&cr->poll, UV_WRITABLE|UV_DISCONNECT, on_poll_connect);
+    struct addrinfo *addr = addrlist;
+    int count = 0;
+    int err = 0;
+    while (addr && count < max_connect_socks) {
```

Comment:
if someone adds 17 IPs and the first 16 all fail i feel like at very least we should tell them that all 16 failed and there were 'n' left we didn't use/try. -- i realize this is quite and unlikely situation...